src/WishlistServices.js

```diff
@@ -0,0 +1,40 @@
+import axios from "axios";
+
+export const handleAddToWishlist=(wishlist,item,encodedToken,dispatch)=>{
+  const isItemPresent = wishlist.find(itemInWishlist => itemInWishlist._id === item._id)
+    if (isItemPresent === undefined) {
+      addToWishlist(item, encodedToken,dispatch)
+    }
+}
+
+export const addToWishlist = (item, encodedToken,dispatch) => {
+   try{
+    fetch("/api/user/wishlist", {
+        method: "POST",
+        body: JSON.stringify({ product: item }),
+        headers: {
+            authorization: encodedToken,
+            "Content-type": "application/json; charset=UTF-8"
+        }
+    })
+        .then(res => res.json())
+        .then(data =>  dispatch({ type: 'SET_WISHLIST', payload: data.wishlist }))
+   }catch(error){
+       console.log(error)
+   }
+}
+
+export const removeFromWishlist=async(id,encodedToken,dispatch)=> {
+    try {
+      const {
+        data: { wishlist },
+      } = await axios.delete(`api/user/wishlist/${id}`, {
```

Comment:
some places you have used axios and some places fetch. Keep it consistent. Use either axios or fetch

Reply:
Will change it soon.